modules/webbrowser/src/processors/webbrowserprocessor.cpp

```diff
@@ -151,7 +151,10 @@ std::string WebBrowserProcessor::getSource() {
 }
 
 WebBrowserProcessor::~WebBrowserProcessor() {
-    static_cast<WebBrowserClient*>(browser_->GetHost()->GetClient().get())->removeLoadHandler(this);
+    WebBrowserClient* webbrowser =
+        static_cast<WebBrowserClient*>(browser_->GetHost()->GetClient().get());
+    webbrowser->removeLoadHandler(this);
+    webbrowser->OnBeforeClose(browser_);
```

Comment:
I don't think that this is a solution to the problem. From the docs of close_browser:
"Results in a call to cef_life_span_handler_t::do_close() if the event handler allows the close or if |force_close| is true (1)"

Reply:
If the code above resolves the issue we might want to look into handling browser destruction in `WebBrowserClient::OnBeforeClose`